src/org.spec.ts

```diff
@@ -255,4 +255,365 @@ describe('org', () => {
+      const nonAsciiAwalaEndpoint = 'はじめよう.みんな';
+      const orgModel = getModelForClass(OrgModelSchema, {
+        existingConnection: connection,
+      });
+      await orgModel.create({
+        name: ORG_NAME,
+        memberAccessType: MemberAccessType.OPEN,
+        awalaEndpoint: AWALA_ENDPOINT,
+      });
+
+      await updateOrg(
+        ORG_NAME,
+        {
+          name: ORG_NAME,
+          memberAccessType: 'INVITE_ONLY',
+          awalaEndpoint: nonAsciiAwalaEndpoint,
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      const dbResult = await orgModel.exists({
+        name: ORG_NAME,
+        memberAccessType: MemberAccessType.INVITE_ONLY,
+        awalaEndpoint: nonAsciiAwalaEndpoint,
+      });
+      expect(dbResult).not.toBeNull();
+    });
+
+    test('Malformed name should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        'INVALID_Name',
+        {
+          name: 'INVALID_Name',
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.MALFORMED_ORG_NAME);
+    });
+
+    test('Non matching name should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        ORG_NAME,
+        {
+          name: `a.${ORG_NAME}`,
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.INVALID_ORG_NAME);
+    });
+
+    test('Malformed Awala endpoint should be refused', async () => {
+      const connection = getConnection();
+
+      const result = await updateOrg(
+        ORG_NAME,
+        {
+          name: ORG_NAME,
+          memberAccessType: 'INVITE_ONLY',
+          awalaEndpoint: 'INVALID_AWALA_ENDPOINT',
+        },
+        {
+          dbConnection: connection,
+          logger: mockLogging.logger,
+        },
+      );
+
+      requireFailureResult(result);
+      expect(result.reason).toBe(OrgProblemType.MALFORMED_AWALA_ENDPOINT);
+    });
+
+    test('Record update errors should be propagated', async () => {
+      const connection = getConnection();
+      await connection.close();
+      const orgData: OrgSchema = {
+        name: ORG_NAME,
+        memberAccessType: 'INVITE_ONLY',
+      };
+
+      const error = await getPromiseRejection(
+        async () =>
+          updateOrg(ORG_NAME, orgData, {
+            dbConnection: connection,
+            logger: mockLogging.logger,
+          }),
+        Error,
+      );
+
+      expect(error).toHaveProperty('name', 'MongoNotConnectedError');
+    });
+
+    describe('getOrg', () => {
+      test('valid name should return the corresponding data', async () => {
```

Comment:
It's not so much that the name is _valid_ what's important, it's that it _exists_.